app/main/db/queries.py

```diff
@@ -221,21 +251,46 @@ def _build_consignment_filter_query(consignment_id: uuid.UUID):
                 ),
                 else_=None,
             ),
-        ),
+        ).label("closure_period"),
     )
 
     filters = [
         File.ConsignmentId == consignment_id,
         func.lower(File.FileType) == "file",
     ]
 
-    query = (
+    sub_query = (
         select.join(FileMetadata, File.FileId == FileMetadata.FileId)
         .filter(*filters)
         .group_by(File.FileId)
         .order_by(File.FileName)
+    ).subquery()
+
+    query = db.session.query(
+        sub_query.c.file_id,
+        sub_query.c.file_name,
+        func.to_char(
+            sub_query.c.date_last_modified,
+            current_app.config["DEFAULT_DATE_FORMAT"],
+        ).label("date_last_modified"),
+        sub_query.c.closure_type,
+        func.to_char(
+            sub_query.c.closure_start_date,
+            current_app.config["DEFAULT_DATE_FORMAT"],
+        ).label("closure_start_date"),
+        sub_query.c.closure_period,
     )
 
+    dt_range = get_date_range(date_range)
+    if date_filter_field is not None:
+        if date_filter_field.lower() == "date_last_modified":
```

Comment:
if we wan to be consistent, we can move this date filtering out into the browse_data function:

just return the unfiltered query here,

then the bottom of browse_data() can become:


```
dt_range = get_date_range(date_range)
if (dt_range["date_from] or dt_range["date_to"]):
    if consignment_id:
        date_filter_query_object = browse_query.c.date_last_modified
    else:
        date_filter_query_object = Consignment.TransferCompleteDatetime

    date_filter = _build_date_range_filter(
        date_filter_query_object,
        dt_range["date_from"],
        dt_range["date_to"],
    )
    browse_query = browse_query.filter(date_filter)

Reply:
I amended the logic for the consignment filter if condition, subquery fields cannot be retrieved in browse_data function so the date filter logic for consignment has to be in the _build_consignment_filter_query function

Reply:
I was hoping we'd get away without the subquery, but yeah if thats necessary can leave this

Reply:
as we compare date in different date format (which db supports) and we show date in different format, we have to keep the sub query as it is.